### PR TITLE
How to Produce Colors With RGB LEDs Using Arduino and Matlab Graphical User Interface

### LINKED ISSUE #6279: How to produce colors with RGB LEDs using arduino and Matlab graphical user interface
#### Introduction
Arduino is an open-source hardware and software company, project, and user community that designs and manufactures single-board microcontrollers and microcontroller kits for building digital devices. Arduino board designs use a variety of microprocessors and controllers. The boards are equipped with sets of digital and analogue input/output (I/O) pins that may be interfaced to various expansion boards ('shields') or breadboards (for prototyping) and other circuits.
This tutorial presents a scheme to produce any colour using RGB LEDs controlled by the Arduino Uno board. The Arduino Uno board is interfaced with MATLAB. Also, one can learn how to interface Arduino Uno with Matlab and get an idea of how different colours are produced from three primary colours Red, Green and Blue.
#### Key takeaways 
- Arduino.
- Hardware Support package for arduino in Matlab.
- How to design the GUI using guide in Matlab.
- Matlab code that produces the colors.

#### Article quality
This article shows on how you can program your arduino using Matlab

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by queenterbruce063**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6279 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @queenterbruce063 ,

Thanks for submitting this PR.

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started.

Meanwhile, we suggest you to go through existing approved PRs/published articles to see if you have missed out anything.

Thanks.

File: content/articles/how-to-produce-colors-with-rgb-leds/index.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+For our case, we have used the common cathode. If you want to switch on the red LED, you apply 5 volts to its terminal/channel. It applies to both green and blue. 
+
+When using the common anode type, we ground the terminal of the color that we want to switch on.
+When you want to produce a red color, you provide 5v to the red terminal. It can be represented as `RGB=[1 0 0]` where 1 represents the maximum voltage, and 0 represents no voltage. It applies to all the colors. 
+
+You have to mix the RGB colors to produce custom colors. A simple combination to produce custom colors is shown below:
+
+![Producing custom colors](/engineering-education/how-to-produce-colors-with-rgb-leds/rgb-four.png)
+
+The image above shows that producing custom colors involves combining the RGB colors. For example, you want to produce a yellow color.
```
Comment (Dawe-7) [inline content/articles/how-to-produce-colors-with-rgb-leds/index.md:75]: The second sentence here is unclear

File: content/articles/how-to-produce-colors-with-rgb-leds/index.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+- Power supply- It is used to provide external power to the Arduino board.
+- Reset- This button is used to reset the Arduino after use.
+- Microcontroller Atmega328- It is the microcontroller. It governs the functionality of the Arduino.
+- Digital I/O- They collect data from the Arduino sensors and write them to other components(actuators)
+- Serial out(Tx)/in(Rx)- It is used to communicate between Arduino and the computer or other connected devices.
+- Analog in- Used for reading the analog sensors.
+
+> In the digital input/output, pins 11, 10, 9, 6, 5, 3 have `~` symbols. This is because they are supported by pulse width modulation(PMW). They can give any voltage in the range `0-5v` with the help of PMW. The pins can generate a dc voltage in 256 variations since it is an 8-bit controller(2^8). Arduino produces this voltage variation by varying the duty cycle of a square wave. It is known as PMW.
+
+In IDE of Arduino. Cc, the voltages of these PMW pins can be written by `analogwrite(pin, value)`. In Matlab, we use the command `writePMWVoltage(board, pin, value)`.
```
Comment (Dawe-7) [inline content/articles/how-to-produce-colors-with-rgb-leds/index.md:46]: The first sentence here is unclear, its missing the determiner `an`. Is Arduino. Cc a manufacturer?

File: content/articles/how-to-produce-colors-with-rgb-leds/index.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+![Producing custom colors](/engineering-education/how-to-produce-colors-with-rgb-leds/rgb-four.png)
+
+The image above shows that producing custom colors involves combining the RGB colors. For example, you want to produce a yellow color. 
+
+However, yellow is formed by a combination of red and green colors. So, we need to switch the red and the green colors simultaneously. It can be represented as `RGB=[1 1 0]`.
+
+At some point, you do not need to provide the full voltage. For example, if you want to produce an orange color, we need the red led to be brighter and green to be less bright. This can be represented as `RGB=[1 0.4 0]`. 
+
+The red led receives 5V, and the green receives 40% of the total voltage. With this variation, you can get any color.
```
Comment (Dawe-7) [inline content/articles/how-to-produce-colors-with-rgb-leds/index.md:81]: Capitaliza `led` throughout the tutorial

File: content/articles/how-to-produce-colors-with-rgb-leds/index.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    set(handles.Board_ON, 'BackgroundColor', [0 1 0]);
+end
+guidata(hObject, handles)
+```
+
+First, we declare the board variable `a = Arduino()`. After this, we create a handle `handle.a` for the board variable. 
+
+We also create a handle for the `ON` condition such that when the handle is 1, the board is on. When the handle is 0, the board is off. It's like a linking handle. This handle is `handles.ON`.
+
+Now, let us use an `if` statement to check for the validity of the conditions. First, we check if the board is created using `~isempty(handles.a)`. If it is created(equal to 1), we set the button's background color to green[0 1 0]. We finally update the handles using `guidata(hObject, handles)`.
```
Comment (Dawe-7) [inline content/articles/how-to-produce-colors-with-rgb-leds/index.md:130]:
```suggestion
Now, let us use an `if` statement to check for the validity of the conditions. First, we check if the board is created using `~isempty(handles.a)`. If it is created (equal to 1), we set the button's background color to green [0 1 0]. We finally update the handles using `guidata(hObject, handles)`.

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
Awesome article @queenterbruce063, kindly implement the suggestions below so that we can move on to tech review. Also, add sources for any images that aren't screenshots and have been downloaded from other sites. Ping me when done.

**Comment by queenterbruce063** [comment]
@Dawe-7 changes made.

**Review by Dawe-7 — APPROVED** [review]
Nice article @queenterbruce063. Kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles. Thank you.

**Pull request merged**